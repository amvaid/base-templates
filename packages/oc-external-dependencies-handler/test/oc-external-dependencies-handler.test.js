const externalDependenciesHandler = require('../index.js');

test('When a module is being used withough being declared in the package', done => {
  const handler = externalDependenciesHandler({});
  const handlerFunction = handler[0];
  handlerFunction(null, 'lodash', err => {
    expect(err).toMatchSnapshot();
    done();
  });
});

test('When a module declared in the package is being used', done => {
  const handler = externalDependenciesHandler({ lodash: '4.17.4' });
  const handlerFunction = handler[0];
  handlerFunction(null, 'lodash', err => {
    expect(err).toBeUndefined();
    done();
  });
});

test('The handler matcher should correctly match aganinst valid modules', () => {
  const handler = externalDependenciesHandler({ lodash: '4.17.4' });
  const handlerMatcher = handler[1];
  expect(handlerMatcher.test('lodash')).toBe(true);
  expect(handlerMatcher.test('yet-another-module')).toBe(true);
  expect(handlerMatcher.test('@org/module-name')).toBe(true);
  expect(handlerMatcher.test('@org/module/path')).toBe(true);
});
