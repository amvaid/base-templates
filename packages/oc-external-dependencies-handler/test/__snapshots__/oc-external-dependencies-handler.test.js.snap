// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When a module is being used withough being declared in the package 1`] = `[Error: Missing dependencies from package.json => "lodash"]`;

exports[`When a module/path is being used withough being declared in the package 1`] = `[Error: Missing dependencies from package.json => "lodash"]`;
